core/src/main/java/org/infinispan/persistence/sifs/IndexNode.java

```diff
@@ -782,11 +783,13 @@ private IndexNode copyWith(int cacheSegment, byte[] indexKey, int file, int offs
             newKeyParts[insertPart] = substring(indexKey, newPrefix.length, keyComp);
             System.arraycopy(leafNodes, 0, newLeafNodes, 0, insertPart + 1);
             System.arraycopy(leafNodes, insertPart + 1, newLeafNodes, insertPart + 2, leafNodes.length - insertPart - 1);
+            log.tracef("Creating new leafNode for %s at %d:%d", objectKey, file, offset);
```

Comment:
Check if trace enabled here and the next added trace too.

Reply:
No need to check if trace enabled for a single line.